backend/src/main/java/neuefische/capstone/bpmndiagram/BpmnDiagramController.java

```diff
@@ -18,29 +18,28 @@ public class BpmnDiagramController {
     private final BpmnDiagramService service;
 
     @GetMapping
-    List<BpmnDiagram> getAllDiagrams(@RequestParam Optional<Boolean> distinct) {
-        if (distinct.orElse(false)) {
+    List<BpmnDiagram> getAllDiagrams(@RequestParam Optional<Boolean> onlylatestversions) {
+        if (onlylatestversions.orElse(false)) {
             return service.getLatestDiagrams();
         } else {
             return service.getAllDiagrams();
         }
     }
 
-    @GetMapping("/history/{key}")
+    @GetMapping("/{key}/history")
     List<BpmnDiagram> getHistoryOfDiagram(@PathVariable String key) {
         return service.getHistoryByKey(key);
     }
 
-
     @PostMapping
     @ResponseStatus(code = HttpStatus.CREATED)
-    public BpmnDiagram addBpmnDiagram(@RequestBody @Valid BpmnDiagram newBpmnDiagram) {
+    BpmnDiagram addBpmnDiagram(@RequestBody @Valid BpmnDiagram newBpmnDiagram) {
         return service.addBpmnDiagram(newBpmnDiagram);
     }
 
-    @PutMapping("/{id}")
-    public BpmnDiagram updateBpmnDiagram(@RequestBody @Valid BpmnDiagram updatedBpmnDiagram, @PathVariable String id) {
-        if (!updatedBpmnDiagram.id().equals(id)) {
+    @PutMapping("/{diagramId}")
+    BpmnDiagram updateBpmnDiagram(@RequestBody @Valid BpmnDiagram updatedBpmnDiagram, @PathVariable String diagramId) {
+        if (!updatedBpmnDiagram.id().equals(diagramId)) {
```

Comment:
If you do not need the PathVariable for anything else, the check can be made obsolete by just handling the RequestBody and saving that